src/main/java/io/rcktapp/api/handler/firehose/FirehoseDb.java

```diff
@@ -45,36 +45,36 @@ public class FirehoseDb extends Db
      */
     protected String includeStreams;
 
-    AmazonKinesisFirehoseAsync firehoseClient = null;
+    protected AmazonKinesisFirehoseAsync firehoseClient = null;
 
     public FirehoseDb() {
         super();
         setType("firehose");
     }
 
     @Override
-    public void bootstrapApi() throws Exception {
-        List<Pair<String, String>> nameActuals = new ArrayList<>();
+    public void bootstrapApi() {
         // our local streams
-        getFirehoseClient().listDeliveryStreams(new ListDeliveryStreamsRequest().withDeliveryStreamType("DirectPut")).getDeliveryStreamNames().forEach(name -> nameActuals.add(Pair.of(name.toLowerCase(), name)));
+        Stream<Pair<String, String>> inKinesis = firehoseNameStream().map(name -> Pair.of(name.toLowerCase(), name));
         // our defined aliases
-        Stream.of(Optional.ofNullable(includeStreams).orElse("").split(",")).map(part -> part.split("\\|")).forEach(arr -> nameActuals.add(Pair.of(arr[0], arr.length > 1 ? arr[1] : arr[0])));
+        Stream<Pair<String, String>> inConfiguration = Stream.of(Optional.ofNullable(includeStreams).orElse("").split(","))
+                .map(part -> part.split("\\|"))
+                .map(arr -> Pair.of(arr[0], arr.length > 1 ? arr[1] : arr[0]));
 
         this.setType("firehose");
-
-        for (Pair<String, String> stream : nameActuals) {
+        Stream.concat(inConfiguration, inKinesis).collect(Collectors.toList()).forEach(stream -> {
```

Comment:
I moved from using for(...) to stream.forEach here, so that I could put the two streams we created already together into one stream. of Pair<String, String>.